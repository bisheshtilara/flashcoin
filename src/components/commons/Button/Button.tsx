import React from "react";
import { CircleLoader } from "react-spinners";

export interface IButtonProps {
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  secondary?: boolean;
  disabled?: boolean;
  loading?: boolean;
  tailwindStyle?: string;
  label?: string;
  type?: "button" | "submit" | "reset" | undefined;
  image?: string;
  imageStyle?: string;
  //Add alt type string later
}

const Button: React.FC<IButtonProps> = ({
  onClick,
  disabled,
  loading,
  tailwindStyle,
  label,
  type = "button",
  secondary,
  image,
  imageStyle,
}) => {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      className={`${tailwindStyle} ${
        secondary ? "border" : "text-white bg-red-600"
      } ${
        image ? "gap-3" : ""
      } w-full rounded-lg h-12 ease-in transition-all hover:-translate-y-1 flex items-center justify-center`}
      type={type}
    >
      {loading ? (
        <CircleLoader color={`${secondary ? "#dc2626" : "#fff"}`} size={35} />
      ) : (
        <>
          {image && (
            <img
              src={image}
              alt="Alt"
              className={`${
                imageStyle ? imageStyle : "h-8 bg-white rounded-full"
              }`}
            />
          )}
          <p>{label}</p>{" "}
        </>
      )}
    </button>
  );
};

export default Button;
