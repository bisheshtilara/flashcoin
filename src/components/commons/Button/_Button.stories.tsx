import { ComponentMeta, ComponentStory } from "@storybook/react";
import Button, { IButtonProps } from "./Button";

export default {
  component: Button,
} as ComponentMeta<typeof Button>;

const enum ButtonType {
  button = "button",
  submit = "submit",
  reset = "reset",
}

const defaultArgs: IButtonProps = {
  onClick: () => alert("Button clicked"),
  secondary: true,
  disabled: false,
  loading: false,
  tailwindStyle:
    "bg-red-600 text-white rounded-full p-3 flex items-center justify-center gap-2 transition-all ease-in hover:-translate-y-2 cursor-pointer",
  label: "Add",
  type: ButtonType.button,
};

const Template: ComponentStory<typeof Button> = (args: IButtonProps) => (
  <Button {...args} />
);

export const Default = Template.bind({});
Default.args = {
  ...defaultArgs,
};
