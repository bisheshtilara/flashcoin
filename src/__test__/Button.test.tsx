import { render, fireEvent, screen } from "@testing-library/react";
import Button from "../components/commons/Button";

describe("Button", () => {
  it("should render correctly", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it("should render a button have border", () => {
    render(<Button secondary />);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("border");
  });

  it("should render a button have text-white bg-red-600", () => {
    render(<Button />);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("text-white bg-red-600");
  });

  it("should render a button have gap-3", () => {
    render(<Button image="https://picsum.photos/200" />);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("gap-3");
  });
});
