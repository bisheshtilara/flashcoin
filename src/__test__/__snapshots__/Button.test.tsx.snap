// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div>
  <button
    class="undefined text-white bg-red-600  w-full rounded-lg h-12 ease-in transition-all hover:-translate-y-1 flex items-center justify-center"
    type="button"
  >
    <p />
     
  </button>
</div>
`;
